hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -95,25 +102,87 @@ spec = parallel $ do
       monitor $ cover 0.1 (null outputs) "no message when reconnecting"
       monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
       monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
-      run . failAfter 5 $ do
+      run . failAfter 15 $ do
         withFreePort $ \port ->
           withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
             mapM_ sendOutput outputs
-            withClient port $ \conn -> do
+            withClient port defaultPath $ \conn -> do
               received <- replicateM (length outputs + 1) (receiveData conn)
               case traverse Aeson.eitherDecode received of
                 Left{} -> failure $ "Failed to decode messages:\n" <> show received
-                Right timedOutputs ->
+                Right timedOutputs -> do
                   (output <$> timedOutputs) `shouldBe` greeting : outputs
 
+  it "does not echo history if client says no" $
+    checkCoverage . monadicIO $ do
+      outputs :: [ServerOutput SimpleTx] <- pick arbitrary
+      monitor $ cover 0.1 (null outputs) "no message when reconnecting"
+      monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
+      monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
+      run . failAfter 5 $ do
+        withFreePort $ \port ->
+          withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
+            -- send arbitrary outputs from the server (which should be ignored by the client)
+            mapM_ sendOutput outputs
```

Comment:
When we call `sendOutput` here, we are neither the client nor the server. It is the hydra-node which is supposed to call this function `sendOutput` to ask the _server_ to send messages to the client.

And this is not clear when we see `sendOutput` here.

I would improve that code to make the context more clear to the reader: who is doing what? Or _who am I_ when I read this code?

As a minimum, I think I would extract a function called something like `hydra_node_sends_messages_to_the_clients`

Reply:
I would say we are the api server when we do `sendOutput`. I will rename the `outputs` to history just to be clear these are the messages we want to skip.